src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncidentMappingResource.kt

```diff
@@ -0,0 +1,144 @@
+package uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.resource
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import jakarta.validation.Valid
+import org.springframework.dao.DuplicateKeyException
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.config.DuplicateMappingErrorResponse
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.config.DuplicateMappingException
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.config.ErrorResponse
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.data.IncidentMappingDto
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.service.IncidentMappingService
+
+@RestController
+@Validated
+@RequestMapping("/mapping/incidents", produces = [MediaType.APPLICATION_JSON_VALUE])
+class IncidentMappingResource(private val mappingService: IncidentMappingService) {
+
+  @PreAuthorize("hasRole('ROLE_NOMIS_INCIDENTS')")
+  @PostMapping("")
+  @ResponseStatus(HttpStatus.CREATED)
+  @Operation(
+    summary = "Creates a new Incident Report mapping",
+    description = "Creates a mapping between a Nomis incident report id and DPS Incident report id. Requires role NOMIS_INCIDENTS",
+    requestBody = io.swagger.v3.oas.annotations.parameters.RequestBody(
+      content = [
+        Content(
+          mediaType = "application/json",
+          schema = Schema(implementation = IncidentMappingDto::class),
+        ),
+      ],
+    ),
+    responses = [
+      ApiResponse(responseCode = "201", description = "Mapping entry created"),
+      ApiResponse(
+        responseCode = "409",
```

Comment:
I think this is duplicated below?

Reply:
will check and remove - think I was a bit copy and paste happy